app/Services/RouteSuggester.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Venue;
+use Fisharebest\Algorithm\Dijkstra;
+use Illuminate\Database\Eloquent\Builder;
+
+class RouteSuggester
+{
+    private float $searchRange = 0.2;
+
+    public function __construct(
+        private array $stopsAttributes,
+        private array $start,
+        private array $end
+    ) {
+    }
+
+    public function suggest()
+    {
+        $stopsVenues =
+            $this->getLocalVenuesByAttributes(
+                $this->stopsAttributes,
+                $this->start,
+                $this->end,
+                $this->searchRange
+            );
+
+        $venuesGraph = $this->createVenuesGraph($stopsVenues);
+        $dijkstra = new Dijkstra($venuesGraph);
```

Comment:
I assume this is using the fisher-yates shuffle algorithm??? Is not a reatively ramdom algo???

Reply:
This is just an off the shelf shortest path algo, I'm not sure how its implemented.

Tbf the algo almost doesn't matter, any graph traversal aglo will do the job. 

Im going to mess around with other graph traversal algos at some point soon.

Reply:
sounds good although I am not sure why we need a graph algo for a list of suggested routes?? Would a 2d array not suffice?

Reply:
I should really write up my thinking here at some point. 

But basically the graph is used to plot a route between the start and end point and make sure each venue that is suggested for each stop gets **progressively closer to the end point**.

A weighted graph and traversal algo is the easiest way to do this I think.

Reply:
How would we use fisher-yates shuffle algorithm in this context?